src/aleph/http/server.clj

```diff
@@ -534,33 +534,41 @@
 
        :channel-read
        ([_ ctx msg]
-         (try
-           (let [ch (.channel ctx)]
-             (if-not (instance? WebSocketFrame msg)
-               (.fireChannelRead ctx msg)
-               (let [^WebSocketFrame msg msg]
-                 (cond
-
-                   (instance? TextWebSocketFrame msg)
-                   (netty/put! ch in (.text ^TextWebSocketFrame msg))
-
-                   (instance? BinaryWebSocketFrame msg)
-                   (let [body (.content ^BinaryWebSocketFrame msg)]
-                     (netty/put! ch in
-                       (if raw-stream?
-                         (netty/acquire body)
-                         (netty/buf->array body))))
-
-                   (instance? PingWebSocketFrame msg)
-                   (netty/write-and-flush ch (PongWebSocketFrame. (netty/acquire (.content msg))))
-
-                   (instance? CloseWebSocketFrame msg)
-                   (.close handshaker ch (netty/acquire msg))
-
-                   :else
-                   (.fireChannelRead ctx msg)))))
-           (finally
-             (netty/release msg)))))]))
+         (let [ch (.channel ctx)]
+           (cond
+             (instance? TextWebSocketFrame msg)
```

Comment:
@kachayev you could use [`(condp instance? msg ...)`](https://clojuredocs.org/clojure.core/condp) to simplify code further.

Reply:
That can be a little excessive :) 

My goal was to get rid of second `fireChannelRead`, and I don't think `condp` is used anywhere in aleph, so just for consistency i would prefer to stick with `cond`

Reply:
If my memory doesn't fail me... `condp` was used for handling websocket frames on the client side. I personally found it harder to read and reason about, but that might be just my personal preference :)